reactjs-films-homework/src/components/Header/Form/__tests__/Form.spec.tsx

```diff
@@ -1,20 +1,34 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
+import { render, screen, cleanup } from '@testing-library/react';
 import Form from '../Form';
+import { Provider } from 'react-redux';
+import store from '../../../../store/store';
 
 describe('Form', () => {
-  render(<Form />);
-  it('Form snapshot', () => {
-    const { asFragment } = render(<Form />);
-    //   screen.debug();
-    expect(asFragment).toMatchSnapshot();
+  let fragment: any;
+
+  beforeEach(() => {
+    const { asFragment } = render(
+      <Provider store={store}>
+        <Form placeholder="Movies" />
+      </Provider>
+    );
+    fragment = asFragment();
+  });
+
+  afterEach(() => {
+    cleanup();
   });
-  it('placeholder text', () => {
-    render(<Form />);
+
+  it('Snapshot', () => {
```

Comment:
It isn't a good name for the test case.
'should renders correctly' is better.
Also, as I see, you can merge these 3 cases into one, because the Form component renders in the same conditions for these cases.

Reply:
done, I rewrote it